lib/http.c

```diff
@@ -517,14 +521,14 @@ CURLcode Curl_http_auth_act(struct connectdata *conn)
   if(conn->bits.user_passwd &&
```

Comment:
@bagder Oh, BTW, shouldn't this be `if((conn->bits.user_passwd || conn->oauth_bearer) &&` now?

Reply:
ah, yes you're right, I believe that would probably be what oauth-bearer users would want!

Reply:
I'll make it so, in a separate commit, but for convenience in the same PR.

Reply:
Done: 9453aac69